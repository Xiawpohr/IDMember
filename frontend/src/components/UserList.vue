<template>
  <v-container fluid grid-list-lg>
    <v-data-iterator
        :items="users"
        :rows-per-page-items="rowsPerPageItems"
        :pagination.sync="pagination"
        content-tag="v-layout"
        row
        wrap
    >
      <v-flex
        slot="item"
        slot-scope="props"
        xs12
        sm6
        md4
        lg3
      >
        <UserItem :user="props.item" />
      </v-flex>
    </v-data-iterator>
  </v-container>
</template>

<script>
import UserItem from './UserItem.vue'

export default {
  components: {
    UserItem
  },
  props: {
    users: {
      type: Array,
      required: true,
      default: () => []
    }
  },
  data() {
    return {
      rowsPerPageItems: [8, 12, 16],
      pagination: {
        rowsPerPage: 8
      },
    }
  }
}
</script>
