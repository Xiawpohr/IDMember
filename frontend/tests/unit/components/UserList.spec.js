import Vue from 'vue'
import Vuetify from 'vuetify'
import { shallowMount } from '@vue/test-utils'
import UserList from '@/components/UserList.vue'

Vue.use(Vuetify)

let wrapper
let users = [{ id: 1, name: 'Arthur' }, { id: 2, name: 'Robert' }]

beforeEach(() => {
  wrapper = shallowMount(UserList, {
    localVue: Vue,
    propsData: {
      users
    }
  })
})

it('should render the component', () => {
  expect(wrapper.html()).toMatchSnapshot()
})
