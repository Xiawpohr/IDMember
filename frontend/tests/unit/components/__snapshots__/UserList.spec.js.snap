// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the component 1`] = `
<vcontainer-stub tag="div" fluid="" grid-list-lg="">
  <vdataiterator-stub contenttag="v-layout" items="[object Object],[object Object]" rows-per-page-items="8,12,16" pagination="[object Object]" row="" wrap=""></vdataiterator-stub>
</vcontainer-stub>
`;
